libs/wire-api/src/Wire/API/EnterpriseLogin.hs

```diff
@@ -232,6 +232,61 @@ instance ToSchema DomainRegistrationResponse where
         <*> (.teamInvite) .= teamInviteObjectSchema
         <*> (.dnsVerificationToken) .= optField "dns_verification_token" (maybeWithDefault Aeson.Null schema)
 
+----------------------------------------------------------------------
+
+data DomainRegistration' = DomainRegistration'
+  { domain :: Domain,
+    settings :: DomainRegistrationSettings',
```

Comment:
Can we make this a `Maybe` and remove the `None` case? I think preauthorized can also be outsize of this, because its not really a setting, but its just a step in configuration. I think the main logic of wire-server (i.e. regsitrations and logins) are not affected by it being pre-authorized, it is treated the same as not having any config.

Reply:
right, domain was a hickup.  and yes, maybe is nicer than none.  :+1:  bd36d474cb4d6e54880b17b888b11b26b7fdbd98

Reply:
on second thought, the code is using the domain field in DomainRegistration a lot, and if we want to replace that with DomainRegistration', the code change in the logic would explode a little, and the new code wouldn't exactly be more concise.

i'll tentatively add the domain back in, let's see if i can convince you.